src/lseq.rs

```diff
@@ -216,6 +216,20 @@ impl<T, A: Ident> LSeq<T, A> {
         Some(op)
     }
 
+    /// Perform a local deletion at `ix`. If `ix` is out of bounds
+    /// then the last element will be deleted, i.e. `self.len() - 1`.
+    pub fn delete_index_or_last(&mut self, ix: usize) -> Op<T, A>
```

Comment:
We could also flip the definitions so that `delete_index` does a length check before calling `delete_index_or_last`. But they're the same to me.